xray/README.md

```diff
@@ -0,0 +1,55 @@
+# X-Ray
+
+A tool to inspect the performance of different wireguard implementations
+
+## How it works
+
+The setup consists of two clients and a wireguard interface. One of the clients sends and receives encrypted packets (let's call this the crypto socket), and the other sends and receives plaintext packets (let's call this the plaintext socket). The packets that are being sent contain a sequence number so that the receiving socket can now which packet it is, and a timestamp of when it was sent. 
+
+Execution consists of these steps:
+1. The wireguard interface is started
+2. `tcpdump` is started, with packets being written to a pcap file
+3. The main X-Ray executable is started
+  3.1. The crypto client performs a handshake with the wireguard interface
+  3.2. The specified number of packets are sent in the specified direction (crypto->wg interface->plaintext, or plaintext->wg interface->crypto)
+  3.3. The receiving socket registers the packets it receives that are part of the test
+  3.4. The packets are written to a `.csv` file with each row containing the sequence number, the time it was sent and, if it was received, the time it was received
+4. `tcpdump` is stopped
+5. The wireguard interface is destroyed
+6. The `.csv` file and pcap are analyzed
+
+## Running it
+
+X-Ray currently only works on linux
+
+Nordlayer and/or nordvpn might interfere with xray, so try disabling them if you're not receving any of the packets
+
+### Support
+
+Supported adapter types:
+- Naitve
+- Neptun
+- WireguardGo (needs to be installed beforehand)
+- Boringtun (needs to be compiled and placed in the `neptun/target/release` folder beforehand)
+
+### Running the test
+
+The application is executed with the `run.py` script. I takes some arguments, all of which are optional:
+
+- **--wg <adapter type>**: which adapter type to use, accepts one of `neptun`, `native`, `wggo` and `boringtun`. Default is `neptun`
```

Comment:
Maybe it makes sense to also allow use of a bin path? to xplatform interface compliant  implementation of wg

Reply:
I think that's generally a good idea though I think there's more work that will need to be done to actually make xray cross platform. I would do this as a separate PR

Reply:
I created LLT-5845 for it